src/llvm/builders/To.js

```diff
@@ -0,0 +1,187 @@
+    // TODO: To use bitmanipulation we have to expose createTrunc and a few missing helpers in llvm-node.
+    let isInfinityOrNaN = builder.createFCmpUEQ(value, ConstantFP.get(llvmContext, Infinity));
+    let isNegativeInfinityOrNaN = builder.createFCmpUEQ(value, ConstantFP.get(llvmContext, -Infinity));
+    let isInfiniteOrNaN = builder.createOr(isInfinityOrNaN, isNegativeInfinityOrNaN);
+    let remainder = builder.createFRem(value, ConstantFP.get(llvmContext, Math.pow(2, 32)));
+    return builder.createSelect(
+      isInfiniteOrNaN,
+      ConstantInt.get(llvmContext, 0),
+      builder.createFPToSI(remainder, LLVMType.getInt32Ty(llvmContext))
+    );
+  } else if (value.type.isIntegerTy(32)) {
+    // Signed Integral
+    return value;
+  } else if (state.intrinsics.isUint32Type(value.type)) {
+    // Unsigned Integral
+    let unsignedValue = builder.createExtractValue(value, [0]);
+    // The bit representation is already correct so we just have to unpack it.
+    return unsignedValue;
+  } else {
+    return buildToInt32(state, buildToNumber(state, value, builder), builder);
+  }
+}
+
+export function buildToUint32(state: CompilerState, value: LLVMValue, builder: IRBuilder): LLVMValue {
+  if (value.type.isIntegerTy(1)) {
+    // Boolean
+    return builder.createSelect(value, ConstantInt.get(llvmContext, 1), ConstantInt.get(llvmContext, 0));
+  } else if (value.type.isDoubleTy()) {
+    // Number
+    // TODO: I think we can make this fewer instructions/faster by some clever bit manipulation.
+    // If not, we should probably hoist this out into a library call instead of inline.
+    // TODO: To use bitmanipulation we have to expose createTrunc and a few missing helpers in llvm-node.
+    let isInfinityOrNaN = builder.createFCmpUEQ(value, ConstantFP.get(llvmContext, Infinity));
+    let isNegativeInfinityOrNaN = builder.createFCmpUEQ(value, ConstantFP.get(llvmContext, -Infinity));
+    let isInfiniteOrNaN = builder.createOr(isInfinityOrNaN, isNegativeInfinityOrNaN);
+    let remainder = builder.createFRem(value, ConstantFP.get(llvmContext, Math.pow(2, 32)));
+    let v = builder.createSelect(
+      isInfiniteOrNaN,
+      ConstantInt.get(llvmContext, 0),
+      builder.createFPToSI(remainder, LLVMType.getInt32Ty(llvmContext))
+    );
+    // We wrap the value in a struct to keep track that this is an unsigned int
+    // so that we know which operations to apply on this.
+    let unsignedValue = UndefValue.get(state.intrinsics.uint32Type);
+    return builder.createInsertValue(unsignedValue, v, [0]);
+  } else if (value.type.isIntegerTy(32)) {
+    // Integral
+    // The bit representation is already correct. We just need to wrap the value
+    // in a struct to keep track that this is an unsigned int so that we know
+    // which operations to apply on this.
+    let unsignedValue = UndefValue.get(state.intrinsics.uint32Type);
+    return builder.createInsertValue(unsignedValue, value, [0]);
```

Comment:
Note how because of the spec is written these things become safe for subsequent operations.